sources/subs/InlinePermissionsForm.class.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+/**
+ * Functions to support the permissions controller
+ *
+ * @name      ElkArte Forum
+ * @copyright ElkArte Forum contributors
+ * @license   BSD http://opensource.org/licenses/BSD-3-Clause
+ *
+ * @version 1.1 beta 1
+ */
+
+/**
+ * Class to initialize inline permissions sub-form and save its settings
+ */
+class Inline_Permissions_Form
+{
+	/**
+	 * Save the permissions of a form containing inline permissions.
+	 *
+	 * @param string[] $permissions
+	 */
+	public static function save($permissions)
+	{
+		global $context;
+
+		$db = database();
+
+		// No permissions? Not a great deal to do here.
+		if (!allowedTo('manage_permissions'))
+			return;
+
+		// Almighty session check, verify our ways.
+		checkSession();
+		validateToken('admin-mp');
+
+		// Make sure they can't do certain things,
+		// unless they have the right permissions.
+		Permissions::loadIllegal();
+
+		$insertRows = array();
+		foreach ($permissions as $permission)
+		{
+			if (!isset($_POST[$permission]))
+				continue;
+
+			foreach ($_POST[$permission] as $id_group => $value)
+			{
+				if (in_array($value, array('on', 'deny')) && (empty($context['illegal_permissions']) || !in_array($permission, $context['illegal_permissions'])))
+					$insertRows[] = array('id_group' => (int) $id_group, 'permission' => $permission, 'add_deny' => $value == 'on' ? 1 : 0);
+			}
+		}
+
+		// Remove the old permissions...
+		$db->query('', '
```

Comment:
Why is redundant? I didn't see it in the function.
https://github.com/elkarte/Elkarte/blob/d210298c7a53f0ed94ecd04b291f4ab0240528f6/sources/subs/ManagePermissions.subs.php#L964